src/test/java/com/google/cloud/tools/gradle/appengine/core/CloudSdkDownloaderTest.java

```diff
@@ -0,0 +1,133 @@
+      downloader.setComponentInstaller(getTempComponentInstaller());
+      downloader.setUpdater(getTempUpdater());
+
+      File sdkHome = downloader.downloadSdk(Version.LATEST.toString());
+
+      Assert.assertTrue(sdkHome.exists());
+      Assert.assertTrue(getTempGcloud().toFile().exists());
+    } catch (IOException | UnsupportedOsException e) {
+      Assert.fail();
+    }
+  }
+
+  @Test
+  public void testDownloadSdk_fixedVersion() {
+    downloader = new CloudSdkDownloader();
+
+    try {
+      downloader.setInstaller(getTempInstaller(new Version("191.0.0")));
+      downloader.setComponentInstaller(getTempComponentInstaller());
+      downloader.setUpdater(getTempUpdater());
+
+      File sdkHome = downloader.downloadSdk("191.0.0");
+
+      Assert.assertTrue(sdkHome.exists());
+      Assert.assertTrue(getTempGcloud().toFile().exists());
+    } catch (IOException | UnsupportedOsException | BadCloudSdkVersionException e) {
+      Assert.fail();
+    }
+  }
+
+  @Test
+  public void testIsSdkValid_true() {
+    downloader = new CloudSdkDownloader();
+
+    try {
+      downloader.setInstaller(getTempInstaller(Version.LATEST));
+      downloader.setComponentInstaller(getTempComponentInstaller());
+      downloader.setUpdater(getTempUpdater());
+
+      File sdkHome = downloader.downloadSdk(Version.LATEST.toString());
+      Assert.assertTrue(downloader.isSdkValid(Version.LATEST.toString(), sdkHome));
+    } catch (IOException | UnsupportedOsException e) {
+      Assert.fail();
+    }
+  }
+
+  @Test
+  public void testIsSdkValid_false() {
+    downloader = new CloudSdkDownloader();
+
+    try {
+      downloader.setInstaller(getTempInstaller(new Version("190.0.0")));
+      downloader.setComponentInstaller(getTempComponentInstaller());
+      downloader.setUpdater(getTempUpdater());
+
+      File sdkHome = downloader.downloadSdk("190.0.0");
+      Assert.assertFalse(downloader.isSdkValid("191.0.0", sdkHome));
+    } catch (IOException | UnsupportedOsException | BadCloudSdkVersionException e) {
+      Assert.fail();
+    }
+  }
+
+  private SdkInstaller getTempInstaller(Version version)
+      throws IOException, UnsupportedOsException {
+    return SdkInstaller.newInstaller(
+        temporaryFolder.newFolder().toPath(),
+        version,
+        OsInfo.getSystemOsInfo(),
+        "google-cloud-tools-java",
+        false);
+  }
+
+  private SdkComponentInstaller getTempComponentInstaller() throws UnsupportedOsException {
+    return SdkComponentInstaller.newComponentInstaller(getTempGcloud());
+  }
+
+  private SdkUpdater getTempUpdater() throws UnsupportedOsException {
+    return SdkUpdater.newUpdater(getTempGcloud());
+  }
+
+  private Path getTempGcloud() throws UnsupportedOsException {
+    OsInfo info = OsInfo.getSystemOsInfo();
+    String gcloudBin = info.name().equals(Name.WINDOWS) ? "gcloud.cmd" : "gcloud";
+    return temporaryFolder.getRoot().toPath().resolve("bin").resolve(gcloudBin);
```

Comment:
It seems that this is resolving a `bin` folder in the temporaryFolder's root - was it supposed to use the `temporaryFolder.newFolder` created by `getTempInstaller`?